src/main/scala/viper/silver/plugin/standard/inline/InlineErrorChecker.scala

```diff
@@ -43,6 +43,31 @@ trait InlineErrorChecker {
     mutRecPreds
   }
 
+  /**
+    * Given a predicate, evaluate to a set containing the names of all predicates called by it.
+    * Only return the names of predicates called that aren't equal to itself.
+    *
+    * @param pred the predicate we want to collect the names of inner predicate calls for.
+    * @return a set of the ids of predicates called in the given predicate.
+    */
+  def nonRecursivePredsCalledBy(pred: Predicate, program: Program, recursivePredNames: Set[String]): Option[Set[String]] =
+    pred.body.map { body =>
+      // Forgive me Father Alonzo for I have Sinned
+      var calledPreds = Set[String]()
+      body.foreach { child =>
+        child.visit {
+          case PredicateAccessPredicate(calledPredAcc, _) =>
+            if (!recursivePredNames(calledPredAcc.predicateName)) {
+              val calledPredLiteral = program.findPredicate(calledPredAcc.predicateName)
+              calledPreds += calledPredAcc.predicateName
+              // I don't like using .getOrElse but it's type-safe in this case
+              calledPreds ++= nonRecursivePredsCalledBy(calledPredLiteral, program, recursivePredNames).getOrElse(Set[String]())
+            }
+        }
+      }
+      calledPreds
+    }
```

Comment:
I think the return type should just be `Set[String]` instead, and you declare `calledPreds` outside of the map, so:
```
var calledPreds = Set[String]()
pred.body.map {
  ...
    calledPreds += calledPredAcc.predicateName
    calledPreds ++= nonRecursivePredsCalledBy(...)
  ...
}
calledPreds
```
No `getOrElse` required :)